androidApp/src/main/java/com/example/tasky/android/login/viewmodel/LoginViewModel.kt

```diff
@@ -0,0 +1,32 @@
+package com.example.tasky.android.login.viewmodel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.tasky.android.login.screen.LoginScreenEvent
+import com.example.tasky.android.login.screen.LoginScreenState
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+class LoginViewModel : ViewModel() {
+    private val _screenStateFlow = MutableStateFlow(LoginScreenState())
+    val screenStateFlow = _screenStateFlow.asStateFlow()
+
+    private val _isLoginSuccessFlow = MutableStateFlow(false)
+    val isLoginSuccessFlow = _isLoginSuccessFlow.asStateFlow()
+
+    fun onEvent(event: LoginScreenEvent) {
+        when (event) {
+            is LoginScreenEvent.OnClickLogin -> login()
+            is LoginScreenEvent.OnClickToSignUp -> {}
+            is LoginScreenEvent.OnEmailStateChange -> _screenStateFlow.value = screenStateFlow.value.copy(emailState = event.newState)
+            is LoginScreenEvent.OnPasswordStateChange -> _screenStateFlow.value = screenStateFlow.value.copy(passwordState = event.newState)
```

Comment:
In general, I'd always use `_screenStateFlow.update { it.copy(...) }` to avoid race conditions. This is how it could happen:

*StateFlow holds value A*
- Thread 1 calls `copy()` on StateFlow with value A
- Before Thread 1 assigns its changed copy, Thread 2 updates StateFlow with value B
- Now, Thread 1 assigns its changed copy, which is still based on value A though, but the latest value is already value B. Therefore, value B would never be considered here.

Not always an issue, but I worked on highly stateful apps with frequently changing states, in which this really bit me, since such bugs are super hard to find.
Using `_screenStateFlow.value = _screenStateFlow.value.copy(...)` is fine if you always call this from a single thread (which you do here). But in ViewModels, you're very likely to not do that as soon as you'd change the state inside a `viewModelScope`.